File: MOBILE/app_sintec/lib/shared/model/questionario.dart
class Questionario{
  int id;
  String titulo;
  int ordemApresentacao;
  int idTipoServico;
  int ativo;

  Questionario({
    required this.id,
    required this.titulo,
    required this.idTipoServico,
    required this.ordemApresentacao,
    required this.ativo
});

  factory Questionario.fromSQLite(Map map) {
    return Questionario(
      id: map["id"],
      titulo: map["titulo"],
      ordemApresentacao: map['ordem_apresentacao'],
      idTipoServico:map['id_tipo_servico'],
      ativo: map["ativo"],
    );
  }

  static List<Questionario> fromSQLiteList(List<Map> listMap) {
    List<Questionario> lista = [];
    for (Map item in listMap) {
      lista.add(Questionario.fromSQLite(item));
    }
    return lista;
  }
}

File: api-sintec/src/main/java/br/com/sintec/model/Questionario.java
package br.com.sintec.model;

import java.time.LocalDateTime;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.JoinColumn;
import javax.persistence.ManyToOne;
import javax.persistence.Table;

import org.springframework.format.annotation.DateTimeFormat;

import com.fasterxml.jackson.annotation.JsonFormat;

import lombok.EqualsAndHashCode;
import lombok.Getter;
import lombok.Setter;
import lombok.ToString;

@ToString
@Getter
@Setter
@Entity
@Table(name="CAD_QUESTIONARIO")
public class Questionario {

	
	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	@EqualsAndHashCode.Include
	@Column(name = "ID")
	private Long id;
	@Column(name = "NM_TITULO")
	private String titulo;
	@Column(name = "NR_ORDEM_APRESENTACAO")
	private int ordemApresentacao;
	@Column(name = "FLAG_OBRIGATORIO")
	private int flagObrigatorio;
	@ManyToOne()
	@JoinColumn(name="ID_TIPO_SERVICO")
	private TipoServico tipoServico;
	@Column(name = "DT_ATUALIZACAO")
	@DateTimeFormat(pattern = "yyyy-MM-dd HH:mm:ss")
	@JsonFormat(pattern = "yyyy-MM-dd HH:mm:ss", shape = JsonFormat.Shape.STRING)
	private LocalDateTime dataAtualizacao;
	@Column(name = "ST_ATIVO")
	private int ativo;
	
	public Questionario(Long id) {
		super();
		this.id = id;
	}

	public Questionario() {
		super();
	}
	
	
	
	
	
	
	
}
